### PR TITLE
test(upgrade): add unit tests for AngularJSUrlCodec's parse method

### PR DESCRIPTION
**Pull request by agale123**
Add unit test coverage for new logic added in #32874 and for existing
logic that was untested.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe: Tests

## What is the current behavior?
N/A

Issue Number: #32874

## What is the new behavior?
Functionality of parse method is tested

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/upgrade/test/params.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      const result = codec.parse('http://server.com:1234/foo?bar=true#heading');
+      expect(result.href).toBe('http://server.com:1234/foo?bar=true#heading');
+      expect(result.protocol).toBe('http');
+      expect(result.host).toBe('server.com:1234');
+      expect(result.search).toBe('bar=true');
+      expect(result.hash).toBe('heading');
+      expect(result.hostname).toBe('server.com');
+      expect(result.port).toBe('1234');
+      expect(result.pathname).toBe('/foo');
+    });
```
Comment (gkalpak) [inline packages/common/upgrade/test/params.spec.ts]: Could you add a couple more testcase:
- URL without `search`.
- URL without `hash`.
- URL with empty path.

File: packages/common/upgrade/test/params.spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    it('should throw an error for an invalid URL', () => {
+      expect(() => {
+        codec.parse('/foo/bar');
+      }).toThrowError('Invalid URL (/foo/bar) with base (undefined)');
+    });
+
+    it('should throw an error for an invalid base', () => {
+      expect(() => {
+        codec.parse('/foo/bar', 'abc');
```
Comment (gkalpak) [inline packages/common/upgrade/test/params.spec.ts]: Could you use a valid URL here, to make it clear that the error comes from the invalid base?

File: packages/common/upgrade/test/params.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      expect(result.port).toBe('1234');
+      expect(result.pathname).toBe('/foo');
+    });
+
+    it('should apply a base', () => {
+      const withoutSlash = codec.parse('foo/bar', 'http://abc.xyz');
+      expect(withoutSlash.href).toBe('http://abc.xyz/foo/bar');
+      const withSlash = codec.parse('/foo/bar', 'http://abc.xyz/');
+      expect(withSlash.href).toBe('http://abc.xyz/foo/bar');
+
```
Comment (gkalpak) [inline packages/common/upgrade/test/params.spec.ts]: Nit: Unnecessary linebreak.

File: packages/common/upgrade/test/params.spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        codec.parse('/foo/bar');
+      }).toThrowError('Invalid URL (/foo/bar) with base (undefined)');
+    });
+
+    it('should throw an error for an invalid base', () => {
+      expect(() => {
+        codec.parse('/foo/bar', 'abc');
+      }).toThrowError('Invalid URL (/foo/bar) with base (abc)');
+    });
+
```
Comment (gkalpak) [inline packages/common/upgrade/test/params.spec.ts]: Nit: Unnecessary linebreak.

**Review by gkalpak — CHANGES_REQUESTED** [review]
This is awesome ✨ Thx so much, @agale123 💯 🥇 ❤️
<sub>People rarely come back to add tests, when they get their bug fix merged :grin:</sub>

I left a couple of suggestions, but otherwise this looks great 👍

File: packages/common/upgrade/test/params.spec.ts (hunk 2 above)
Comment (agale123) [reply packages/common/upgrade/test/params.spec.ts]: Done

File: packages/common/upgrade/test/params.spec.ts (hunk 1 above)
Comment (agale123) [reply packages/common/upgrade/test/params.spec.ts]: Added a few more. Focused on testing the parts near search/hash for those test cases

**Review by gkalpak — APPROVED** [review]
LGTM 👍
Thx again, @agale123 ✨

**Comment by gkalpak** [comment]
Oh, one last thing:
All source files must to have a license header. To make the linter happy, can you please add the following comment at the top of `params.spec.ts`:

https://github.com/angular/angular/blob/fca4be5207fb534fbaf47f083a5d0a2287ef4515/packages/common/upgrade/test/upgrade.spec.ts#L1-L8

**Comment by agale123** [comment]
> Oh, one last thing:
> All source files must to have a license header. To make the linter happy, can you please add the following comment at the top of `params.spec.ts`:
> 
> https://github.com/angular/angular/blob/fca4be5207fb534fbaf47f083a5d0a2287ef4515/packages/common/upgrade/test/upgrade.spec.ts#L1-L8

Oh thanks, yeah I just added that.

**Pull request closed without merging**